components/browser/search/src/main/java/mozilla/components/browser/search/suggestions/Parser.kt

```diff
@@ -0,0 +1,97 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.search.suggestions
+
+import mozilla.components.browser.search.SearchEngine
+import org.json.JSONArray
+import org.json.JSONObject
+
+/*
+ * Convenience method to convert a JSONArray into a sequence.
+ */
+private fun JSONArray.asSequence(): Sequence<Any> {
+    return object : Sequence<Any> {
+
+        override fun iterator() = object : Iterator<Any> {
+            val it = (0 until this@asSequence.length()).iterator()
+
+            override fun next(): Any {
+                val i = it.next()
+                return this@asSequence.get(i)
+            }
+
+            override fun hasNext() = it.hasNext()
+        }
+    }
+}
```

Comment:
This is nice. I think we should move this into `support-ktx` and make it public. :)

Reply:
👍